File: SumofTwoIntegers.py
class Solution:
    def getSum(self, a, b):
        """
        :type a: int
        :type b: int
        :rtype: int
        """
        MAX = 0x7FFFFFFF
        MIN = 0x80000000
        mask = 0xFFFFFFFF

        while b != 0:
            a, b = (a ^ b) & mask, ((a & b) << 1) & mask

        return a if a <= MAX else ~(a ^ mask)


if __name__ == '__main__':
    a = int(input())
    b = int(input())
    solution = Solution()
    print(solution.getSum(a, b))

File: SumofTwoIntegers.cpp
#include<iostream>

using namespace std;

class Solution {
public:
    int getSum(int a, int b) {
        
        if(b == 0) return a;
        
        int sum = a ^ b;
        int carry = (a & b) << 1;
        
        return getSum(sum, carry);
        
    }
};

int main() {
	
	int a, b;
	cin>>a>>b;
	
	Solution *solution = new Solution();
	cout<<solution->getSum(a, b);
	
	return 0;
	
}
